examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +153,43 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
-
+pid_t pid;
+int ret_exec;
+int wait_status;
+int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+if (fd < 0) { perror("open"); abort(); }
+switch (pid = fork()) {
```

Comment:
@SrirajVemparala think about how you could reduce duplication with the function above.